src/libstd/sys/unix/os.rs

```diff
@@ -257,6 +258,28 @@ pub fn current_exe() -> io::Result<PathBuf> {
     }
 }
 
+#[cfg(any(target_os = "sunos"))]
+pub fn current_exe() -> io::Result<PathBuf> {
+    extern {
+        fn getexecname() -> *const c_char;
+    }
+    unsafe {
+        let path = getexecname();
+        if path.is_null() {
+            Err(io::Error::last_os_error())
+        } else {
+            let filename = CStr::from_ptr(path).to_bytes();
+            if filename[0] == b'/' {
+                Ok(PathBuf::from(<OsStr as OsStrExt>::from_bytes(filename)))
```

Comment:
You may be able to lift these `&[u8] -> &OsStr` conversions outside of the `if` branch as it happens in both branches